src/Data/UndefinedOr.purs

```diff
@@ -81,43 +90,43 @@ toUndefined = UndefinedOr
 -- | Like `maybe` but for undefined
 runUndefined :: forall a b. b -> (a -> b) -> UndefinedOr a -> b
 runUndefined b f u@(UndefinedOr a) =
-    if isUndefined u
-        then b
-        else f a
-
+  if isUndefined u then b
+  else f a
 
 instance eqUndefined :: Eq a => Eq (UndefinedOr a) where
-    eq ux@(UndefinedOr x) uy@(UndefinedOr y) = case unit of
-      _ | isUndefined ux && isUndefined uy -> true
-        | not (isUndefined ux) && not (isUndefined uy) -> eq x y
-      _ -> false
-
-instance functorUndefined :: Functor UndefinedOr where
-    map f u@(UndefinedOr x) =
-        if isUndefined u
-            then undefinedVal
-            else UndefinedOr (f x)
-
-instance applyUndefined :: Apply UndefinedOr where
-    apply uf@(UndefinedOr f) ux@(UndefinedOr x) =
-        if isUndefined uf || isUndefined ux
-            then undefinedVal
-            else UndefinedOr (f x)
-
-instance applicativeUndefined :: Applicative UndefinedOr where
-    pure = toUndefined
-
-instance altUndefined :: Alt UndefinedOr where
-    alt x y =
-        if isUndefined x
-            then y
-            else x
-
-instance plusUndefined :: Plus UndefinedOr where
-    empty = undefinedVal
+  eq ux@(UndefinedOr x) uy@(UndefinedOr y) = case unit of
+    _
+      | isUndefined ux && isUndefined uy -> true
+      | not (isUndefined ux) && not (isUndefined uy) -> eq x y
+    _ -> false
+
+-- | unlawful version of `map` - caution does not obey the functor laws
```

Comment:
Nitpick: can you capitalize the comments and fix the punctuation? By fixing I mean `s/ - caution/. Caution:/`